### ISSUE TITLE
user.type helper is slow even with delay 0

### ISSUE THREAD

**Issue by ald890** [testing-library/user-event#857]
Description:
Version: 14.0.0-beta.11

Tested two variants for 100 characters text.
I use default config with delay=0

1. `userEvent.click` and `userEvent.paste`- 70ms~
```
await user.click(screen.getByLabelText(/label/));
await user.paste(textMock);
```

2. `userEvent.type` - 1500ms~
```
await user.type(screen.getByLabelText(/label/), textMock);
```

`keypress` - seems to be slow, because there is a sync loop without Promise.all() and each iteration has 10-15ms

[This fragment of code](https://github.com/testing-library/user-event/blob/892e6d569a2dd994947942235f4ecbbd411c3fe4/src/keyboard/keyboardAction.ts#L49) is synchronous and slow 

```javascript
 if (unpreventedDefault && hasKeyPress(keyDef, config)) {
        await keypress(keyDef, getCurrentElement, config)
}
```

**Comment by ph-fritsche** [comment]
`paste()` is a completely different abstraction. It pastes the content at once.

`type()` simulates pressing and releasing keys.

They are completely different and comparing them is moot.

This looks like a duplicate of #577.

**Comment by ald890** [comment]
I know that, but `type()` with delay = 0, should be faster

**Comment by ph-fritsche** [comment]
So what is the acceptable execution time and how did you estimate it?

As mentioned on the other issue:
> If you want to demonstrate the performance of the implementation, you need to record the events it dispatches and replay those calling all the same event handlers.

>If you can provide a reproduction, we'll look into it.
If you keep on comparing the execution time to something completely different, we'll get nowhere.

It is very likely that this is a problem with your environment.

**Comment by ald890** [comment]
Acceptable execution time should be <100ms because in another case we can get Jest timeout, because if I use 5x `type()` then the test will take <5s it is too long for one simple test case.

In my local env test on 8core CPU test are quite fast 5s, but on CI they have event worst perfromance.

**Comment by ph-fritsche** [comment]
I'll close this issue, as there it lacks a reproduction example.

This looks like this is a problem with the environment.
When we `await` we allow other code to run in between - the next microtask with `delay: null` or the next macrotask with `delay: 0`. It might be the DOM implementation, the event handlers or any other code on the event loop causing the excessive execution time.

**Comment by ald890** [comment]
This is not true... each `keypress` has a delay of 10-15ms, so if you have "Lorem ipsum", then you have to wait 100ms this is ridicules

**Comment by ph-fritsche** [comment]
> This is not true...

Which part is not true? There is no reproduction showing any problem with our implementation - neither on this issue nor on the other one which this one is a duplicate of.

If you can create a reproduction example that showcases a problem on our end, I'll gladly reopen.
